dev-lang/fpc/fpc-3.2.2.ebuild

```diff
@@ -0,0 +1,160 @@
+
+	# let the pkg manager compress man files
+	sed -i '/find man.* gzip /d' "${WORKDIR}"/fpcbuild-"${PV}"/install/man/Makefile || die
+
+	# make the compiled binary check for fpc.cfg under the prefixed /etc/ path
+	hprefixify "${WORKDIR}"/fpcbuild-"${PV}"/fpcsrc/compiler/options.pas
+}
+
+set_pp() {
+	case "${ARCH}" in
+		amd64)	FPC_ARCH="x64" FPC_PARCH="x86_64" ;;
+		x86)	FPC_ARCH="386" FPC_PARCH="i386" ;;
+		*)	die "This ebuild doesn't support ${ARCH}." ;;
+	esac
+
+	case ${1} in
+		bootstrap)	pp="${WORKDIR}/lib/fpc/${PV_BIN}/ppc${FPC_ARCH}" ;;
+		new) 	pp="${S}/compiler/ppc${FPC_ARCH}" ;;
+		*)	die "set_pp: unknown argument: ${1}" ;;
+	esac
+}
+
+src_compile() {
+	local pp
+
+	# Using the bootstrap compiler.
+	set_pp bootstrap
+
+	emake PP="${pp}" compiler_cycle AS="$(tc-getAS)"
+
+	# Save new compiler from cleaning...
+	cp "${S}/compiler/ppc${FPC_ARCH}" "${S}/ppc${FPC_ARCH}.new" || die
+
+	# ...rebuild with current version...
+	emake PP="${S}/ppc${FPC_ARCH}.new" AS="$(tc-getAS)" compiler_cycle
+
+	# ..and clean up afterwards
+	rm "${S}/ppc${FPC_ARCH}.new" || die
+
+	# Using the new compiler.
+	set_pp new
+
+	emake PP="${pp}" AS="$(tc-getAS)" rtl_clean
+
+	# ide is moved to packages directory and build unconditionaly.
+	emake PP="${pp}" AS="$(tc-getAS)" rtl packages_all utils
+}
+
+src_install() {
+	local pp
+	set_pp new
+
+	#fpcbuild-3.0.0/utils/fpcm/fpcmake
+	#${WORKDIR}/${PN}build-${PV}/utils/fpcm/fpcmake"
+	#fpcbuild-3.0.0/fpcsrc/utils/fpcm/bin/x86_64-linux/fpcmake
+	set -- PP="${pp}" FPCMAKE="${S}/utils/fpcm/bin/${FPC_PARCH}-linux/fpcmake" \
+		INSTALL_PREFIX="${ED}/usr" \
+		INSTALL_DOCDIR="${ED}/usr/share/doc/${PF}" \
+		INSTALL_MANDIR="${ED}/usr/share/man" \
+		INSTALL_SOURCEDIR="${ED}/usr/lib/fpc/${PV}/source"
+
+	emake "$@" compiler_install rtl_install packages_install utils_install
+
+	dosym ../lib/fpc/"${PV}"/ppc"${FPC_ARCH}" /usr/bin/ppc"${FPC_ARCH}"
+
+	cd "${S}"/../install/doc || die
+	emake "$@" installdoc
+
+	cd "${S}"/../install/man || die
+	emake "$@" installman
+
+	if use doc ; then
+		cd "${S}"/../../doc || die
+		dodoc -r *
+	fi
+
+	if use source ; then
+		cd "${S}" || die
+		shift
+		emake PP="${ED}/usr/bin/ppc${FPC_ARCH}" "$@" sourceinstall
+		find "${ED}/usr/lib/fpc/${PV}/source" -name '*.o' -exec rm {} \;
+	fi
+
+	"${ED}/usr/lib/fpc/${PV}/samplecfg" "${ED}/usr/lib/fpc/${PV}" "${ED}"/etc || die
+
+	# set correct (prefixed) path for e.g. unit files
+	sed -i "s:${ED}:${EPREFIX}:g" "${ED}"/etc/fpc.cfg || die
+
+	sed -e "s:${ED}::g" \
+		-i "${ED}/etc/fppkg.cfg" \
+		-i "${ED}/etc/fppkg/default" \
+		-i "${ED}/usr/lib/fpc/${PV}/ide/text/fp.cfg" \
+		|| die
+
+	rm -r "${ED}"/usr/lib/fpc/lexyacc || die
+
+	case "${ARCH}" in
+		amd64)
+			mkdir -p "${ED}/usr/$(get_libdir)/"
+			mv "${ED}"/usr/lib/*.so "${ED}/usr/$(get_libdir)/" || die ;;
+	esac
+}
+
+pkg_postinst() {
+	if [[ -z "${REPLACING_VERSIONS}" ]] ; then
+		einfo "To read the documentation in the fpc IDE, enable the doc USE flag"
+	fi
+}
```

Comment:
I know of lot of these were already in the old ebuild, but einfo has very little visibility unless user stares at the build log (for a one time message, elog is typically preferable). Also shouldn't it not show if the doc USE is already enabled? But I question the need of this message altogether.

**If** you wish to clarify that USE=doc has runtime implications, then adding a local definition in metadata.xml may make more sense. Users may often look at use descriptions when something is amiss.

Reply:
Add condition to check USE="-doc". That use flag installs common html documentation as well, so reading it from ide is feature in my mind rather than runtime option.